pkg/analyze/download.go

```diff
@@ -188,29 +186,15 @@ func parseAnalyzers(spec string) ([]*troubleshootv1beta2.Analyze, error) {
 	}
 
 	// SupportBundle overwrites Analyzer if defined
-	if gvk.String() == "troubleshoot.sh/v1beta2, Kind=SupportBundle" {
+	if gvk.Group == "troubleshoot.sh" && gvk.Version == "v1beta2" && gvk.Kind == "SupportBundle" {
 		supportBundle := obj.(*troubleshootv1beta2.SupportBundle)
-		analyzer := supportBundleToAnalyzer(supportBundle)
-		return analyzer.Spec.Analyzers, nil
-	} else {
+		return supportBundle.Spec.Analyzers, nil
+	} else if gvk.Group == "troubleshoot.sh" && gvk.Version == "v1beta2" && gvk.Kind == "Analyzer" {
 		analyzer := obj.(*troubleshootv1beta2.Analyzer)
 		return analyzer.Spec.Analyzers, nil
 	}
-}
-
-func supportBundleToAnalyzer(sb *troubleshootv1beta2.SupportBundle) *troubleshootv1beta2.Analyzer {
-	return &troubleshootv1beta2.Analyzer{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "troubleshoot.sh/v1beta2",
-			Kind:       "Analyzer",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name: fmt.Sprintf("%s-analyzer", sb.Name),
-		},
-		Spec: troubleshootv1beta2.AnalyzerSpec{
-			Analyzers: sb.Spec.Analyzers,
-		},
-	}
+	
+	return nil, errors.Wrapf(err, "invalid analyzer %q", gvk)
```

Comment:
maybe
```
errors.Wrapf(err, "invalid gvk %q", gvk)
```

Reply:
`errors.Errorf` is probably what we want here :)

Reply:
@divolgin updated